src/main/java/org/broadinstitute/hellbender/utils/read/GATKRead.java

```diff
@@ -0,0 +1,255 @@
+package org.broadinstitute.hellbender.utils.read;
+
+import com.google.api.services.genomics.model.Read;
+import htsjdk.samtools.Cigar;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.util.Locatable;
+
+import java.util.UUID;
+
+/**
+ * Unified read interface for use throughout the GATK.
+ *
+ * This is the read-only subset of possible operations. Use {@link MutableGATKRead} if you need to modify
+ * reads in-place.
+ */
+public interface GATKRead extends Locatable {
```

Comment:
Either at the interface level or at the method level define whether the caller can perform modifying operations on the read and what happens if they do. Also, specify if the underlying object can change. Ie, once I have a GATKRead, can it change from under me (ie can a method on a GATKRead returns something and then the same method return something else some other time?)